voorhoede/head-start
# feature: XML sitemap and robots.txt

# Changes

- Generates XML sitemaps for all pages, based on `getStaticPaths()` and using the official [`@astro/sitemap`](https://docs.astro.build/en/guides/integrations-guide/sitemap/) package.
- Adds a basic `robots.txt` file.
- Adds a reference to the generated XML sitemap in the `robots.txt` with the deployment's site URL.

# Associated issue

Resolves #5 

# How to test

1. Open preview link
2. Navigate to `/sitemap-index.xml`
3. Verify the sitemap is generated and correctly links to sub sitemaps
4. Navigate to the sub sitemap from the sitemap index (`/sitemap-0.xml`)
5. Verify it correctly lists all available pages (check that one exists for example)
6. Navigate to `/robots.txt`
7. Verify it correctly has a rule pointing to the sitemap index
8. Navigate to any HTML page on the website
9. Verify that it has a `link[rel="sitemap"]` correctly pointing to the sitemap index

# Checklist

- ~~I have added/updated tests to prove that my feature works (if not possible please explain why)~~
- ~~I have made changes to the README and if the change affects the project setup (npm commands changed, new service added, environmental variable added)~~
- ~~I have added a decision log entry if the change affects the architecture or changes a significant technology~~
